### PR TITLE
feat!: rename association classes to prevent conflicts with decorators

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR proposes the following renames:

- `BelongsTo` -> `BelongsToAssociation`
- `BelongsToMany` -> `BelongsToManyAssociation`
- `HasOne` -> `HasOneAssociation`
- `HasMany` -> `HasManyAssociation`

This is a breaking change (needs a documentation PR & a special commit message)

I think this change would be interesting due to the fact that the association decorators share the same name as the association classes.

I gave the shorter name to the decorators as they are used much more often than the association classes themselves

### CODE REVIEW

**Comment by ephys** [comment]
I don't know why this fails now but it should be fixed by https://github.com/sequelize/sequelize/pull/16512

File: packages/core/src/associations/has-many.ts
Comment (WikiRik) [inline packages/core/src/associations/has-many.ts:1]: Line 36 should be updated to use `{@link HasOneAssociation}` I think

**Review by WikiRik — COMMENTED** [review]
https://github.com/sequelize/sequelize/blob/main/packages/core/src/index.mjs should also be updated to reflect the new names

**Comment by WikiRik** [comment]
Can we add some tests that fail if we decided to revert this change? So that future maintainers don't bite themselves in the foot if they do so

**Comment by ephys** [comment]
I added a comment above the class to prevent changing the name, but I can add a test too if you'd prefer

File: packages/core/src/associations/has-many.ts
Comment (ephys) [reply packages/core/src/associations/has-many.ts:1]: Yep! Done

**Pull request merged**